cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -97,4 +112,39 @@ protected ScheduledExecutorService executor() {
     executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("heartbeats-scheduler"));
     return executor;
   }
+
+  /**
+   * Get the size of events ingested by a stream since its creation, in bytes.
+   * @param streamId id of the stream
+   * @return Size of events ingested by a stream since its creation
+   * @throws IOException when getting an error retrieving the metric
+   */
+  protected long getStreamEventsSize(Id.Stream streamId) throws IOException {
+    MetricDataQuery metricDataQuery = new MetricDataQuery(
```

Comment:
the question about metric ttl was more around if it's possible for the number to go down all of a sudden because some data has expired.  Might not be the case for counter metrics?